src/runtime_src/hip/core/stream.h

```diff
@@ -5,21 +5,74 @@
 
 #include "context.h"
 
+#include <condition_variable>
+#include <list>
+
 namespace xrt::core::hip {
 
-// stream_handle - opaque stream handle
-using stream_handle = void*;
+// forward declarations
+class event;
+class command;
 
 class stream
 {
   std::shared_ptr<context> m_ctx;
+  unsigned int m_flags;
+  bool null;
+
+  std::list<std::shared_ptr<command>> cmd_queue;
```

Comment:
Just curious about the motivation to use a `std::list`.

Reply:
std::list was used because I need a container where I can insert/remove elements at both ends (front and back) and also I need to remove elements from middle.